service/frontend/http_api_server.go

```diff
@@ -327,6 +334,11 @@ func (h *HTTPAPIServer) incomingHeaderMatcher(headerName string) (string, bool)
 	if h.matchAdditionalHeaders[headerName] {
 		return headerName, true
 	}
+	for _, prefix := range h.matchAdditionalHeaderPrefixes {
+		if strings.HasPrefix(headerName, prefix) {
```

Comment:
This could be optimized with a trie. Thoughts if it's needed? The number of prefixes is likely going to be small i.e. 1-3 of them

Reply:
No need to optimize this IMO